src/Robo/RoboFileAbstract.php

```diff
@@ -1,12 +1,41 @@
 <?php
+
 namespace ShoppinPal\YapepCommon\Robo;
 
 use Exception;
+use InvalidArgumentException;
 use josegonzalez\Dotenv\Loader;
 use Robo\Tasks;
+use ShoppinPal\YapepCommon\Robo\Swagger\DocumentationGenerator;
+use ShoppinPal\YapepCommon\Robo\Swagger\OpenApiGenerator;
+use ShoppinPal\YapepCommon\Robo\Swagger\SwaggerGenerator;
 
 abstract class RoboFileAbstract extends Tasks
 {
+    protected const SWAGGER_VERSION_V2 = 'v2';
+    protected const SWAGGER_VERSION_V3 = 'v3';
+
+    /**
+     * @return ApplicationConfig[]
+     */
+    abstract protected function getSwaggerConfigs(): array;
```

Comment:
Not sure if i like us calling it swagger, openapi may be a better name as they changed their name officially somewhere between v2 and 3